### ISSUE TITLE
Non alphanumeric strings not allowed in types.identifier(types.string)

### ISSUE THREAD

**Issue by ssured** [mobxjs/mobx-state-tree#158]
See http://jsbin.com/sucucoyoya/5/edit?html,js,console for some minimal code showing the issue

From the set `["coffee", "cof$fee", "cof|fee", "cof/fee"]` only the first is allowed as an identifier string. I don't know if that is intended and thus needs documentation, or if this is a bug.

This seems related to #88 

Thanks for sharing this awesome library (from Groningen)!

**Comment by ssured** [comment]
Copy of the thrown errors:

```
"cof$fee"
"[mobx-state-tree] Error while converting `{\"id\":\"cof$fee\",\"title\":\"Get coffee\"}` to `Todo`:
at path \"/id\" value `\"cof$fee\"` is not assignable to type: `string` (The provided identifier is not valid)."
"cof|fee"
"[mobx-state-tree] Error while converting `{\"id\":\"cof|fee\",\"title\":\"Get coffee\"}` to `Todo`:
at path \"/id\" value `\"cof|fee\"` is not assignable to type: `string` (The provided identifier is not valid)."
"cof/fee"
"[mobx-state-tree] Error while converting `{\"id\":\"cof/fee\",\"title\":\"Get coffee\"}` to `Todo`:
at path \"/id\" value `\"cof/fee\"` is not assignable to type: `string` (The provided identifier is not valid)."
```

**Comment by mattiamanzati** [comment]
Should be fixed once #152 gets merged! :)
